webapp/components/interop-feature-chart.js

```diff
@@ -0,0 +1,215 @@
+    chart.draw(dataTable, this.getChartOptions(div, feature));
+  }
+
+  getChromeChangelogUrl(fromVersion, toVersion) {
+    // Strip off the 'dev' suffix if there.
+    fromVersion = fromVersion.split(' ')[0];
+    toVersion = toVersion.split(' ')[0];
+    return `https://chromium.googlesource.com/chromium/src/+log/${fromVersion}..${toVersion}?pretty=fuller&n=10000`;
+  }
+
+  getFirefoxStableChangelogUrl(fromVersion, toVersion) {
+    // The version numbers are reported as XX.Y.Z, but pushlog wants
+    // 'FIREFOX_XX_Y_Z_RELEASE'.
+    const fromParts = fromVersion.split('.');
+    const fromRelease = `FIREFOX_${fromParts.join('_')}_RELEASE`;
+    const toParts = toVersion.split('.');
+    const toRelease = `FIREFOX_${toParts.join('_')}_RELEASE`;
+    return `https://hg.mozilla.org/mozilla-unified/pushloghtml?fromchange=${fromRelease}&tochange=${toRelease}`;
+  }
+
+  clickFirefoxNightlyDialogGoButton() {
+    const fromSha = this.$.firefoxNightlyDialogFrom.value;
+    const toSha = this.$.firefoxNightlyDialogTo.value;
+    const url = `https://hg.mozilla.org/mozilla-unified/pushloghtml?fromchange=${fromSha}&tochange=${toSha}`;
+    window.open(url);
+  }
+
+  getChartOptions(containerDiv, feature) {
+    // Show only the scores from this year on the charts.
+    // The max date shown on the X-axis is the end of this year.
+    const year = parseInt(this.year);
+    const maxDate = new Date(year + 1, 0, 1);
+    const ticks = [];
+    for (let month = 0; month < 12; month++) {
+      // Show month ticks in the middle of the month on the graph (15th day).
+      ticks.push(new Date(year, month, 15));
+    }
+    const focusAreas = this.getYearProp('focusAreas');
+    const summaryFeatureName = this.getYearProp('summaryFeatureName');
+    if (feature !== summaryFeatureName && !(feature in focusAreas)) {
+      feature = summaryFeatureName;
+    }
+    const options = {
+      height: 350,
+      fontSize: 14,
+      tooltip: {
+        trigger: 'both',
+      },
+      hAxis: {
+        format: 'MMM',
+        viewWindow: {
+          max: maxDate
+        },
+        ticks: ticks,
+        slantedText: true,
+        slantedTextAngle: 90,
+        showTextEvery: 1,
+        gridlines: {
+          count: 13,
+        }
+      },
+      vAxis: {
+        format: 'percent',
+        viewWindow: {
+          min: 0,
+          max: 1,
+        }
+      },
+      explorer: {
+        actions: ['dragToZoom', 'rightClickToReset'],
+        axis: 'horizontal',
+        keepInBounds: true,
+        maxZoomIn: 4.0,
+      },
+      // Line chart color definitions for [Chrome, Firefox, Safari, Interop].
+      colors: ['#fbc013', '#fc7a3a', '#148cda', '#2c6f07'],
+    };
+
+    options.width = '100%';
+    options.legend = {
+      position: 'top',
+      alignment: 'center',
+    };
+    options.chartArea = {
+      left: 75,
+      width: '80%',
+    };
+
+    return options;
+  }
+}
+export { InteropFeatureChart };
```

Comment:
Can you add a new line to the end of each of these files?

Reply:
Fixed!